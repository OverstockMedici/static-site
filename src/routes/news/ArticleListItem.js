import React from 'react';
import dayjs from 'dayjs';
import { Article } from './news.style';

const ArticleListItem = ({ link, imageURL, title, createdAt, paragraphs }) => {
    const buildParagraphs = (paragraphs) =>
        paragraphs.map((paragraph, i) => (
            // eslint-disable-next-line react/no-array-index-key
            <p key={i}>{paragraph}</p>
        ));

    return (
        <Article data-testid="article-list">
            {/* <a href={link} target="_blank" rel="noopener noreferrer">
                <img
                    data-testid="article-list-image"
                    src={imageURL}
                    alt={title}
                />
            </a> */}
            <a
                className="article-heading"
                href={link}
                target="_blank"
                rel="noopener noreferrer"
            >
                <h2>{title}</h2>
            </a>
            {buildParagraphs(paragraphs)}
            <a
                className="read-more-link"
                href={link}
                target="_blank"
                rel="noopener noreferrer"
            >
                Read the full article
            </a>
            <time>{dayjs(createdAt).format('MMM DD[,] YYYY')}</time>
        </Article>
    );
};

export default ArticleListItem;
