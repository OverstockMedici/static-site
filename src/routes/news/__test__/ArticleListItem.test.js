import React from 'react';
import ReactDOM from 'react-dom';
import dayjs from 'dayjs';
import ArticleListItem from './../ArticleListItem';

import { render, cleanup } from '@testing-library/react';
import { getByAltText } from '@testing-library/dom';

import renderer from 'react-test-renderer';

afterEach(cleanup);

const listProps = {
    id: 1,
    link: 'https://stripe.com/',
    imageURL: 'https://stripe.com/img/v3/home/social.png/',
    title: 'Stripe',
    createdAt: 1585288800000,
    paragraphs: [
        'SettleMint, a Belgium-based blockchain',
        'With its scalable low-code, infrastructure-agnostiâ€¦sations that want to adopt and deploy blockchain.',
    ],
};
const { id, link, imageURL, title, createdAt, paragraphs } = listProps;

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <ArticleListItem
            {...{ id, link, imageURL, title, createdAt, paragraphs }}
        />,
        div,
    );
});

it('renders ArticleList correctly', () => {
    const { getByTestId, container } = render(
        <ArticleListItem
            {...{ id, link, imageURL, title, createdAt, paragraphs }}
        />,
    );

    const articleListComponent = getByTestId('article-list');

    expect(articleListComponent).toHaveTextContent(paragraphs[0]);
    expect(articleListComponent).toHaveTextContent(title);
    expect(articleListComponent).toHaveTextContent(
        dayjs(createdAt).format('MMM DD[,] YYYY'),
    );
});

it('matches snapshot', () => {
    const articleListTree = renderer.create(
        <ArticleListItem
            {...{ id, link, imageURL, title, createdAt, paragraphs }}
        />,
    ).toJSON();

    expect(articleListTree).toMatchSnapshot();
});
