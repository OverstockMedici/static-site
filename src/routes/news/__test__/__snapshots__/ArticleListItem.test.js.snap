// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<article
  className="newsstyle__Article-xbmq8k-2 hIeTAB"
  data-testid="article-list"
>
  <a
    href="https://stripe.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <img
      alt="Stripe"
      data-testid="article-list-image"
      src="https://stripe.com/img/v3/home/social.png/"
    />
  </a>
  <a
    className="article-heading"
    href="https://stripe.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <h2>
      Stripe
    </h2>
  </a>
  <p>
    SettleMint, a Belgium-based blockchain
  </p>
  <p>
    With its scalable low-code, infrastructure-agnostiâ€¦sations that want to adopt and deploy blockchain.
  </p>
  <a
    className="read-more-link"
    href="https://stripe.com/"
    rel="noopener noreferrer"
    target="_blank"
  >
    Read the full article
  </a>
  <time>
    Mar 27, 2020
  </time>
</article>
`;
